packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx

```diff
@@ -54,60 +57,62 @@ export const RichTextBlock = memo(function RichTextBlock({
-        <FormatIconButton icon="list-ol" format={BlockFormat.OrderedList} />
-
-        <ToolbarDivider />
-
-        <EditorSubMenuButton icon="table" editorHasFocus={hasFocus}>
-          <TableMenu />
-        </EditorSubMenuButton>
-
-        <ToolbarDivider />
-
-        <FormatIconButton icon="bold" format={TextFormat.Bold} />
-        <FormatIconButton icon="italic" format={TextFormat.Italic} />
-        <FormatIconButton icon="underline" format={TextFormat.Underline} />
-        <FormatIconButton icon="strikethrough" format={TextFormat.Strikethrough} />
-        <FormatIconButton icon="superscript" format={TextFormat.Superscript} />
-        <FormatIconButton icon="subscript" format={TextFormat.Subscript} />
-
-        <ToolbarDivider />
-
-        <LinkFormatButton />
-        <RemoveLinkFormatButton />
-
-        <ToolbarDivider />
-
-        <SubMenuButton icon="smile-o">
-          <EmojiPicker setEmoji={emoji => editor.insertText(emoji)} />
-        </SubMenuButton>
-      </Toolbar>
+      {!displayOnly && (
+        <Toolbar
+          fadeOut={!hasFocus}
+          onMouseDown={e => {
+            e.preventDefault()
+            if (!hasFocus && location) focusAtPreviousLocation(location)
+          }}>
+          <FormatButton format={BlockFormat.H1}>
+            <H1Icon />
+          </FormatButton>
+          <FormatButton format={BlockFormat.H2}>
+            <H2Icon />
+          </FormatButton>
+          <FormatButton format={BlockFormat.H3}>
+            <H3Icon />
+          </FormatButton>
+
+          <ToolbarDivider />
+
+          <FormatIconButton icon="list-ul" format={BlockFormat.UnorderedList} />
+          <FormatIconButton icon="list-ol" format={BlockFormat.OrderedList} />
+
+          <ToolbarDivider />
+
+          <EditorSubMenuButton icon="table" editorHasFocus={hasFocus}>
+            <TableMenu />
+          </EditorSubMenuButton>
+
+          <ToolbarDivider />
+
+          <FormatIconButton icon="bold" format={TextFormat.Bold} />
+          <FormatIconButton icon="italic" format={TextFormat.Italic} />
+          <FormatIconButton icon="underline" format={TextFormat.Underline} />
+          <FormatIconButton icon="strikethrough" format={TextFormat.Strikethrough} />
+          <FormatIconButton icon="superscript" format={TextFormat.Superscript} />
+          <FormatIconButton icon="subscript" format={TextFormat.Subscript} />
+
+          <ToolbarDivider />
+
+          <LinkFormatButton />
+          <RemoveLinkFormatButton />
+
+          <ToolbarDivider />
+
+          <SubMenuButton icon="smile-o">
+            <EmojiPicker setEmoji={emoji => editor.insertText(emoji)} />
+          </SubMenuButton>
+        </Toolbar>
+      )}
       {WepublishEditor.isEmpty(editor) && ( // Alternative placeholder
         <div onClick={() => ReactEditor.focus(editor)} style={{color: '#cad5e4'}}>
           {t('blocks.richText.startWriting')}
         </div>
       )}
```

Comment:
Think the conditional hiding should include the placeholder, as it states 'start writing' ...

Reply:
Fixed :ok_hand: